rasa/utils/tensorflow/rasa_layers.py

```diff
@@ -0,0 +1,423 @@
+import tensorflow as tf
+from typing import Text, List, Dict, Any, Union, Optional, Tuple
+import tensorflow_addons as tfa
+
+from rasa.utils.tensorflow.model_data import FeatureSignature
+from rasa.utils.tensorflow.constants import (
+    REGULARIZATION_CONSTANT,
+    WEIGHT_SPARSITY,
+    NUM_TRANSFORMER_LAYERS,
+    TRANSFORMER_SIZE,
+    NUM_HEADS,
+    UNIDIRECTIONAL_ENCODER,
+    KEY_RELATIVE_ATTENTION,
+    VALUE_RELATIVE_ATTENTION,
+    MAX_RELATIVE_POSITION,
+    HIDDEN_LAYERS_SIZES,
+    DROP_RATE,
+    SPARSE_INPUT_DROPOUT,
+    DENSE_INPUT_DROPOUT,
+    DENSE_DIMENSION,
+    CONCAT_DIMENSION,
+    DROP_RATE_ATTENTION,
+    SEQUENCE,
+    SENTENCE,
+)
+from rasa.utils.tensorflow import layers
+from rasa.shared.nlu.constants import TEXT
+
+# TODO: use this? it's in layers.py
+tfa.options.TF_ADDONS_PY_OPS = True
+
+
+class ConcatenateSparseDenseFeatures(tf.keras.layers.Layer):
+    # TODO add docstring
+    def __init__(
+        self,
+        attribute: Text,
+        feature_type: Text,
+        data_signature: List[FeatureSignature],
+        dropout_rate: float,
+        sparse_dropout: bool,
+        dense_dropout: bool,
+        dense_concat_dimension: int,
+        sparse_to_dense_kw: Dict[Text, Any] = {},
```

Comment:
empty dict is not a valid default value

Reply:
what is this dict?

Reply:
> is not a valid default value

What do you mean? Not valid in what sense?

The dict can have any parameters accepted by `DenseForSparse`. Would you pass these parameters differently?

Reply:
let's use standard `**kwargs` or explicitly mention the params

Reply:
Well... That way we introduce an additional dependency and we'll have to change `rasa_layers` whenever something in `DenseForSparse` changes. Do you think that's not big enough of an issue?

Reply:
Alright, turns out I didn't properly understand how kwargs work. I'm changing this to use them.